diff --git a/ChangeLog b/ChangeLog
index 42eb02a..fddbfb0 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,10 @@
+2008-03-12  Aurelien Francillon <aurelien.francillon@inrialpes.fr>
+
+	[1833340] cannot read program counter/Watchdog has expired
+	* src/jtag2run.cc: Fix jtag2::eventLoop, the function needs to
+	actually loop plus shoudl set breakpoint = true only when event
+	EVT_BREAK is received 
+ 
 2008-02-25  Joerg Wunsch <j.gnu@uriah.heep.sax.de>
 
 	* src/devdescr.cc: Add entries for ATtiny84 and ATtiny85.
diff --git a/src/jtag2run.cc b/src/jtag2run.cc
index 221e5e8..bef6dcb 100644
--- a/src/jtag2run.cc
+++ b/src/jtag2run.cc
@@ -113,6 +113,8 @@ bool jtag2::resumeProgram(void)
 
 bool jtag2::eventLoop(void)
 {
+  for(;;){
+    
     int maxfd;
     fd_set readfds;
     bool breakpoint = false, gdbInterrupt = false;
@@ -170,7 +172,7 @@ bool jtag2::eventLoop(void)
 			  gdbInterrupt = true;
 			  printf("\nIDR dirty: 0x%02x\n", evtbuf[9]);
 		      }
-                    else
+                    else if (evtbuf[8] == EVT_BREAK)
 		      {
 			  breakpoint = true;
 		      }
@@ -184,6 +186,7 @@ bool jtag2::eventLoop(void)
 	return false;
     if (breakpoint)
 	return true;
+  }
 }
 
 
