diff -Nura avr-libc-1.2.3/common/macros.inc avr-libc-1.2.3.crossdev/common/macros.inc
--- avr-libc-1.2.3/common/macros.inc	2004-11-01 18:19:54.000000000 -0300
+++ avr-libc-1.2.3.crossdev/common/macros.inc	2005-05-31 16:11:36.000000000 -0300
@@ -35,6 +35,9 @@
      Created by Marek Michalkiewicz <marekm@linux.org.pl>
  */
 
+#ifndef _MACROS_INC
+#define _MACROS_INC
+
 #include <avr/io.h>
 
 /* if not defined, assume old version with underscores */
@@ -219,3 +222,4 @@
 #endif
 	.endm
 
+#endif
