diff --git a/src/avrora/sim/types/SensorSimulation.java b/src/avrora/sim/types/SensorSimulation.java
index 5740671..66b8d74 100644
--- a/src/avrora/sim/types/SensorSimulation.java
+++ b/src/avrora/sim/types/SensorSimulation.java
@@ -230,6 +230,7 @@ public class SensorSimulation extends Simulation {
                     updateVariable(smap, "node_address", id);               // SOS
                     updateVariable(smap, "TOS_NODE_ID", id);                // Tinyos 2.0
                     updateVariable(smap, "ActiveMessageAddressC$addr", id); // Tinyos 2.0
+                    updateVariable(smap, "ActiveMessageAddressC__addr", id); // Tinyos 2.0 witout $$ in identifiers 
                 }
             }
         }
