--- tools/tinyos/misc/tos-locate-jre-orig	2006-11-03 20:44:39.000000000 +0100
+++ tools/tinyos/misc/tos-locate-jre	2006-11-03 21:25:20.000000000 +0100
@@ -50,6 +50,12 @@
     ;;
 
     Linux)
+
+    if [ -f /etc/gentoo-release ]; then
+	jhome=`java-config -o`/bin
+	echo $jhome
+	exit 0
+    fi
     # We check the path first, on the assumption that that's the preferred
     # version.
     pathlocate javac || { test -z "$javac" && pathlocate java; }
@@ -104,7 +110,9 @@
     # Look for a likely JNI directory
     # Windows, and IBM Java: in jre/bin
     # Sun Java on Linux: in jre/lib/i386
-    if "$jhome/bin/java" -version 2>&1 | grep -q IBM || cygpath -w / >/dev/null 2>/dev/null; then
+    if [ -f /etc/gentoo-release ]; then
+	dir=`java-config -o`/include
+    elif "$jhome/bin/java" -version 2>&1 | grep -q IBM || cygpath -w / >/dev/null 2>/dev/null; then
 	jnilocate "$jhome/jre/bin" || jnilocate "$jhome/bin"
     else
 	arch=`uname -m`
